test/strategy/test_onnx.py

```diff
@@ -0,0 +1,252 @@
+        self.tokenizer = AutoTokenizer.from_pretrained(model_name)
+        self.sequence_a = "intel-extension-for-transformers is based in SH"
+        self.sequence_b = "Where is intel-extension-for-transformers based? NYC or SH"
+        self.encoded_dict = self.tokenizer(self.sequence_a, self.sequence_b, return_tensors="pt")
+        self.encoded_dict["labels"] = 1
+        self.batch_size = 1
+
+    def __iter__(self):
+        yield {
+            "input_ids": self.encoded_dict["input_ids"].detach().cpu().numpy(),
+            "attention_mask": self.encoded_dict["attention_mask"].detach().cpu().numpy(),
+        }, self.encoded_dict["labels"]
+
+
+class TestWeightOnlyQuantTuning(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        cmd = "optimum-cli export onnx --model hf-internal-testing/tiny-random-gptj --task text-generation gptj/"
+        p = subprocess.Popen(
+            cmd, preexec_fn=os.setsid, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True
+        )  # nosec
+        p.communicate()
+
+        self.model = onnx.load("gptj/decoder_model.onnx")
+        self.dataloader = DummyNLPDataloader("hf-internal-testing/tiny-random-gptj")
+
+    @classmethod
+    def tearDownClass(self):
+        shutil.rmtree("nc_workspace", ignore_errors=True)
+        shutil.rmtree("gptj", ignore_errors=True)
+
+    def test_common(self):
+        conf = PostTrainingQuantConfig(
+            approach="weight_only",
+        )
+        q_model = quantization.fit(self.model, conf)
+        for data, _ in self.dataloader:
+            q_out = Inference(q_model.model, data)
+            org_out = Inference(self.model, data)
+            for q, org in zip(q_out, org_out):
+                self.assertTrue((np.abs(q_out[0] - org_out[0]) < 0.5).all())
+
+    def test_basic(self):
+        conf = PostTrainingQuantConfig(
+            approach="weight_only",
+            quant_level=1,
+            op_type_dict={
+                ".*": {  # re.match
+                    "weight": {
+                        "bits": 8,  # 1-8 bits
+                        "group_size": -1,  # -1 (per-channel)
+                        "scheme": "sym",
+                    },
+                },
+            },
+        )
+
+        from neural_compressor.strategy.utils.constant import WoqTuningParams
+
+        acc_res_lst = [1.0] * len(WoqTuningParams)
+        acc_res_lst[2] = 1.2
+        acc_res_lst = [1.3] + acc_res_lst + [1.2, 1.3]
+
+        def fake_eval(model):
+            acc = acc_res_lst.pop(0)
+            return acc
+
+        q_model = quantization.fit(self.model, conf, calib_dataloader=self.dataloader, eval_func=fake_eval)
+        for data, _ in self.dataloader:
+            q_out = Inference(q_model.model, data)
+            org_out = Inference(self.model, data)
+            for q, org in zip(q_out, org_out):
+                self.assertTrue((np.abs(q_out[0] - org_out[0]) < 0.5).all())
+
+        # set algorithm
```

Comment:
Suggest separating the below test into an independent test or writing a common `_test_woq_tuning_xxx` function that accepts different user configs.